### DISCUSSION TITLE
Patching PLUMED with Nequip

### DISCUSSION THREAD

**Discussion by nisargjoshi10** [mir-group/pair_nequip#35] [Q&A] [answered]
Hi all,

Has anyone tried patching PLUMED in LAMMPS with the pair_nequip style? I've been following LAMMPS [documentation](https://docs.lammps.org/Build_extras.html#plumed) but getting 
` Could NOT find FFMPEG (missing: FFMPEG_EXECUTABLE)` and ` Could NOT find BLAS (missing: BLAS_LIBRARIES)` error. I've tried both CMakeBuild and Traditional make methods.

Wondering if I am missing some additional lines in my `CMakeLists.txt` or somewhere else. I would appreciate any suggestions to resolve this.

Thanks,
Nisarg

**Comment by anjohan** [comment]
Hi Nisarg,

These issues sound unrelated to NequIP. Could you try to compile a version of LAMMPS without NequIP to confirm this?

(And of course, make sure you actually have the modules it's complaining about. I'm guessing FFMPEG is warning, not an error.)

**Comment by nisargjoshi10** [comment, marked as answer]
Hi @anjohan,

Thanks for your reply. Yeah it was unrelated to NequIP, I was curious if anybody had tried and figured out a documentation for getting plumed patch with `pair_nequip` installation.

I figured out and posting the instructions here in case anybody needs it. The rest of the instructions for compiling LAMMPS are pretty much the same as mentioned in README. I just had to add a plumed flag and paths to some of my libraries which the code was failing to read 

```
 cmake ../cmake -DCMAKE_PREFIX_PATH=../libtorch -DCUDNN_LIBRARY_PATH=/gscratch/pfaendtner/nisargj/codes/miniconda/envs/nequip/lib -DCUDNN_INCLUDE_PATH=/gscratch/pfaendtner/nisargj/codes/miniconda/envs/nequip/include -DMKL_INCLUDE_DIR=/gscratch/pfaendtner/nisargj/codes/miniconda/envs/nequip/include -D PKG_PLUMED=yes -DBLAS_LIBRARIES=/gscratch/pfaendtner/nisargj/codes/BLAS-3.11.0 -DFFMPEG_LIBRARIES=/gscratch/pfaendtner/nisargj/codes/ffmpeg-5.1.1-amd64-static -DLAPACK_LIBRARIES=/gscratch/pfaendtner/nisargj/codes/lapack-3.11 -DGSL_INCLUDE_DIR=/gscratch/pfaendtner/nisargj/codes/gsl-2.7.1/include -DGSL_LIBRARY=/gscratch/pfaendtner/nisargj/codes/gsl-2.7.1/lib/libgsl.so -DGSL_CBLAS_LIBRARY=/gscratch/pfaendtner/nisargj/codes/gsl-2.7.1/lib/libgslcblas.so
```

Only the `cmake` command changes here and its followed by `make -j{n_proc}`

Hope its helpful to others!

Cheers,
Nisarg
